pilot/pkg/bootstrap/server.go

```diff
@@ -1281,3 +1288,73 @@ func (s *Server) initStatusManager(_ *PilotArgs) {
 		return nil
 	})
 }
+
+// initLocalSecretListener creates a new secret listener to listen for
+// secrets which represent new multi cluster participating clusters in a mesh.
+func (s *Server) initLocalSecretListener(args *PilotArgs) {
+	// Check if this is an instance of Istiod running as external istiod
+	// The logic is based on the following:
+	// 1. command istioctl secret create will always create a secret named
+	//    istio-kubeconfig when secret type is config, to indicate that a cluster
+	//    gets used as a config cluster, the secret type must be config
+	// 2. this secret is required to be added to the namespace where istiod will be
+	//    running, otherwise, istiod wont run as an external istiod
+	// 3. this secret has to be mounted to a location to which the env KUBECONFIG points
```

Comment:
Can't KUBECONFIG be used for completely different purposes too, so using it doesn't mean it's a config cluster for an external control plane?